var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -97,87 +97,66 @@ def setup_run_environment(self, env):
-        )
-
-        filename = "testfloat_8_8_128.dat.sz"
-        decfile = test_data_dir.join(filename)
-
-        reason = "testing 2D decompression of {0}".format(exe)
-        options = ["-x", "-f", "-i", orifile, "-s", decfile, "-2", "8", "1024", "-a"]
-
-        self.run_test(
-            exe,
-            options,
-            [],
-            installed=True,
-            purpose=reason,
-            skip_missing=True,
-            work_dir=test_data_dir,
-        )
-
-    def _test_3d_float(self):
+        with working_dir(test_data_dir):
+            filename = "testfloat_8_8_128.dat"
+            orifile = test_data_dir.join(filename)
+            exe = which("sz")
+            if exe is None:
+                raise SkipTest(f"sz is not installed for version {self.version}")
+
+            with test_part(self, "test_2d_compression_sz", purpose="testing 2D compression of sz"):
+                options = ["-z", "-f", "-i", orifile, "-M", "REL", "-R", "1E-3", "-2", "8", "1024"]
+                exe(*options)
+
+            filename = "testfloat_8_8_128.dat.sz"
+            decfile = test_data_dir.join(filename)
+
+            with test_part(
+                self, "test_2d_decompression_sz", purpose="testing 2D decompression of sz"
+            ):
+                options = ["-x", "-f", "-i", orifile, "-s", decfile, "-2", "8", "1024", "-a"]
+                exe(*options)
+
+    def test_3d_float(self):
         """This test performs simple 3D compression/decompression (float)"""
 
         test_data_dir = self.test_suite.current_test_data_dir
-
-        filename = "testfloat_8_8_128.dat"
-        orifile = test_data_dir.join(filename)
-
-        exe = "sz"
-        reason = "testing 3D compression of {0}".format(exe)
-        options = ["-z", "-f", "-i", orifile, "-M", "REL", "-R", "1E-3", "-3", "8", "8", "128"]
-
-        self.run_test(
-            exe,
-            options,
-            [],
-            installed=True,
-            purpose=reason,
-            skip_missing=True,
-            work_dir=test_data_dir,
-        )
-
-        filename = "testfloat_8_8_128.dat.sz"
-        decfile = test_data_dir.join(filename)
-
-        reason = "testing 3D decompression of {0}".format(exe)
-        options = ["-x", "-f", "-i", orifile, "-s", decfile, "-3", "8", "8", "128", "-a"]
-
-        self.run_test(
-            exe,
-            options,
-            [],
-            installed=True,
-            purpose=reason,
-            skip_missing=True,
-            work_dir=test_data_dir,
-        )
-
-    def test(self):
-        """Perform smoke tests on the installed package"""
-        # run 2D compression and decompression (float)
-        self._test_2d_float()
-        # run 3D compression and decompression (float)
-        self._test_3d_float()
+        with working_dir(test_dat_dir):
+
+            filename = "testfloat_8_8_128.dat"
+            orifile = test_data_dir.join(filename)
+            exe = which("sz")
```

Comment:
Same issue regarding ensuring the executable that is found is the installed one (i.e., one under the prefix).